game/yodesk.ksy

```diff
@@ -0,0 +1,581 @@
+    enums:
+      hotspot_type:
+        0: drop_quest_item
+        1: spawn_location
+        2: drop_unique_weapon
+        3: vehicle_to
+        4: vehicle_back
+        5: drop_map
+        6: drop_item
+        7: npc
+        8: drop_weapon
+        9: door_in
+        10: door_out
+        11: unused
+        12: lock
+        13: teleporter
+        14: ship_to_planet
+        15: ship_from_planet
+  zone_auxiliary:
+    seq:
+      - id: marker
+        contents: "IZAX"
+      - id: size
+        type: u4
+      - id: unknown_count
+        type: u2
+      - id: monster_count
+        type: u2
+      - id: monsters
+        type: monster
+        repeat: expr
+        repeat-expr: monster_count
+      - id: required_item_count
+        type: u2
+      - id: required_items
+        type: u2
+        repeat: expr
+        repeat-expr: required_item_count
+      - id: goal_item_count
+        type: u2
+      - id: goal_items
+        type: u2
+        repeat: expr
+        repeat-expr: goal_item_count
+  zone_auxiliary_2:
+     seq:
+      - id: marker
+        contents: "IZX2"
+      - id: size
+        type: u4
+      - id: provided_item_count
+        type: u2
+      - id: provided_items
+        type: u2
+        repeat: expr
+        repeat-expr: provided_item_count
+
+  zone_auxiliary_3:
+     seq:
+      - id: marker
+        contents: "IZX3"
+      - id: size
+        type: u4
+      - id: npc_count
+        type: u2
+      - id: npc
+        type: u2
+        repeat: expr
+        repeat-expr: npc_count
+  zone_auxiliary_4:
+     seq:
+      - id: marker
+        contents: "IZX4"
+      - id: size
+        type: u4
+      - id: unknown
+        type: u2
+  zones:
+    seq:
+      - id: zone_count
+        type: u2
+      - id: zones
+        type: zone
+        repeat: expr
+        repeat-expr: zone_count
+  puzzles:
+    seq:
+      - id: size
+        type: u4
+      - id: puzzles
+        type: puzzle
+        repeat: until
+        repeat-until: _.index == -1
+  puzzle:
+    seq:
+      - id: index
+        type: s2
+      - id: marker
+        if: index != -1
+        contents: "IPUZ"
+      - id: size
+        type: u4
+        if: index != -1
+      - id: type
+        type: u4
+        if: index != -1
+      - id: unknown1
+        type: u4
+        if: index != -1
+      - id: unknown2
+        type: u4
+        if: index != -1
+      - id: unknown3
+        type: u2
+        if: index != -1
+      - id: strings
+        type: prefixed_str
+        repeat: expr
+        repeat-expr: 5
+        if: index != -1
+      - id: item_1
+        type: u2
+        if: index != -1
+      - id: item_2
+        type: u2
+        if: index != -1
+  endf:
+    seq:
+      - id: empty
+        type: u4
+  characters:
+    seq:
+      - id: size
+        type: u4
+      - id: characters
+        type: character
+        repeat: until
+        repeat-until: _.index == -1
+  character:
+    seq:
+      - id: index
+        type: s2
```

Comment:
Do other "negative" indices make sense? Or are they positive and just 0xFFFF is special?

Reply:
Throughout the whole file signed integers could be considered unsigned with 0xFF FF indicating an empty value. The original game treats alls values expect u4 as signed integers, often checking `x < 0` to determine if an id should be resolvable or if an index marks the end of an array. 

So I think one could argue that the file format contains unsigned values treating 0xFFFF as a special case and the original game has a _bug_ in its parsing code.